internal/charmstore/store.go

```diff
@@ -1425,6 +1425,115 @@ func (store *Store) Search(sp SearchParams) (SearchResult, error) {
 	return result, nil
 }
 
+var listFilters = map[string]string{
+	"name":        "name",
+	"owner":       "user",
+	"series":      "serties",
+	"type":        "type",
+	"promulgated": "promulgated-revision",
+}
+
+func prepareList(sp SearchParams) (filters map[string]interface{}, sort []string, err error) {
+	if len(sp.Text) > 0 {
+		return nil, nil, errgo.New("text not allowed")
+	}
+	if sp.Limit > 0 {
+		return nil, nil, errgo.New("limit not allowed")
+	}
+	if sp.Skip > 0 {
+		return nil, nil, errgo.New("skip not allowed")
+	}
+	if sp.AutoComplete {
+		return nil, nil, errgo.New("autocomplete not allowed")
+	}
+
+	filters = make(map[string]interface{})
+	for k, v := range sp.Filters {
+		switch k {
+		case "name":
+			filters[k] = v[0]
+		case "owner":
+			filters["user"] = v[0]
+		case "series":
+			filters["series"] = v[0]
+		case "type":
+			if v[0] == "bundle" {
+				filters["series"] = "bundle"
+			} else {
+				filters["series"] = map[string]interface{}{"$ne": "bundle"}
+			}
+		case "promulgated":
+			if v[0] != "0" {
+				filters["promulgated-revision"] = map[string]interface{}{"$gt": 0}
```

Comment:
shouldn't this be $ge ? if promulgated-revision = 0 then it's still promulgated.